### DISCUSSION TITLE
split profiling into multiple JFR files

### DISCUSSION THREAD

**Discussion by 1v0dev** [async-profiler/async-profiler#559] [Q&A] [answered]
HI, I have a question. If you run profiling for an extended period, the size of the JFR file becomes quite big. Do you have a plan to add a way to split profiling to write to multiple JFR files, for example if the file grows beyond some size, it closes it and creates a new one?

**Comment by apangin** [comment, marked as answer]
There is already `loop` option that may help.
From the [documentation](https://github.com/jvm-profiling-tools/async-profiler/blob/master/README.md):

> * `--loop TIME` - run profiler in a loop (continuous profiling).
>   The argument is either a clock time (`hh:mm:ss`) or
>   a loop duration in `s`econds, `m`inutes, `h`ours, or `d`ays.
>   Make sure the filename includes a timestamp pattern, or the output
>   will be overwritten on each iteration.  
>   Example: `./profiler.sh --loop 1h -f /var/log/profile-%t.jfr 8983`
> 

It does almost what you want, except that `jfr` rotation happens after the specified time rather than the file size.

**Reply by 1v0dev** [reply]
Thanks, that is great. Time based also works for me.
